src/java/org/apache/cassandra/index/sasi/utils/trie/AbstractPatriciaTrie.java

```diff
@@ -790,7 +790,8 @@ TrieEntry<K, V> nextEntryImpl(TrieEntry<K, V> start, TrieEntry<K, V> previous, T
      * This is implemented by going always to the left until
      * we encounter a valid uplink. That uplink is the first key.
      */
-    TrieEntry<K, V> firstEntry()
+    // @Override needed in JDK 21+.
```

Comment:
if `@Override` is needed, then why don't you add it?

Reply:
Because it is needed only if we build on JDK21+. There is nothing to override on previous version and we still build on 11.